heat/core/manipulations.py

```diff
@@ -1724,6 +1724,10 @@ def reshape(a: DNDarray, shape: Tuple[int, ...], new_split: Optional[int] = None
     if not isinstance(a, DNDarray):
         raise TypeError("'a' must be a DNDarray, currently {}".format(type(a)))
 
+    # check for nested sequence
+    if isinstance(shape[0], (tuple, list)):
```

Comment:
Is `tuple` and `list` enough here? Why not accept `array_like`? It that case, this could be replaced by 
`if not np.isscalar(shape[0]):`

Reply:
Thanks @ben-bou , I introduced the change you suggested, replaced most of the checks with `try...except` statements and am now allowing for `array_like`